pkg/cmd/roachtest/cluster.go

```diff
@@ -2199,12 +2199,30 @@ func (c *clusterImpl) StartE(
 			return err
 		}
 	}
+	nodes := selectedNodesOrDefault(opts, c.CRDBNodes())
+	// N.B. If `SkipInit` is set, we don't wait for SQL since node(s) may not join the cluster in any definite time.
+	if !startOpts.RoachprodOpts.SkipInit {
+		// Wait for SQL to be ready on all nodes, for 'system' tenant, only.
```

Comment:
Should something similar be added for the virtual cluster start counterpart: `StartServiceForVirtualClusterE`?

Reply:
Ah, I guess it's not needed because we're just mainly waiting for the cache to be ready which is at the system level?

Reply:
Technically, yes, we do need to wait for shared-process tenants. However, there could be any number of them, e.g.,

```
I241212 11:19:11.321291 53045 sql/syntheticprivilegecache/cache.go:198 â‹® [T3,Vmixed-version-tenant-mbqzz,n3] 145  warmed privileges for virtual tables in 10.007996435s
```

I thought it's probably best to tackle them in the mixedversion framework, since the above code already feels heavy wrt additional waiting. What do you think?

Reply:
>However, there could be any number of them, e.g.

But don't you just need to wait for the tenant you are starting? Which it should already know from `startOpts.RoachprodOpts.VirtualClusterName`?

>since the above code already feels heavy wrt additional waiting

Yeah I was thinking the same thing, especially since the first SQL check is redundant if `WaitForReplicationFactor > 0` since `WaitForReplication` is also checking if SQL is ready. No arguments against doing it in the mixed version framework here.

Reply:
> But don't you just need to wait for the tenant you are starting?

Technically, yes. Although, in the case of shared-process tenants, they are _automatically_ restarted by system. The case of shared-process is already handled in the mixedversion framework, via `waitForSharedProcess`. The case of separate-process is also _technically_ handled albeit indirectly. Inside `SyncedCluster.Start`, after starting a _separate_ process, `createAdminUserForSecureCluster` essentially ensures SQL is initialized upon return.

Having said that, mostly as a precaution, I repurposed `waitForSharedProcess` as `waitForTenantProcess` which is now used for each deployment type, other than system.